src/state.ts

```diff
@@ -24,62 +24,129 @@ export type TransitionEvent<TContext = unknown, TEvent = unknown> = (
 	hookInput: HookInput<TContext, TEvent>,
 ) => boolean | Promise<boolean>;
 
-export type GenericInitialState<TContext = unknown, TEvent = unknown> =
-	| InitialGuardState
-	| InitialAutoTransitionState;
-
-export interface InitialGuardState<TContext = unknown, TEvent = unknown> {
+export interface InitialState<TContext = unknown, TEvent = unknown> {
 	id: StateIdentifier;
-	initial: boolean;
+	initial: true;
+	transitionTo: TransitionToHook<TContext, TEvent>;
+
+	autoTransition?: false;
 	transitionGuard?: TransitionEvent<TContext, TEvent>;
+}
+
+export interface InitialWithFinalFalseState<
```

Comment:
what's the point of having `final: false`, the property should not be in the type

Reply:
So you think that the best for the user is to have `final: true` or not have the property `final` ?! so the user can't specify `final: false` ?